advisor/backend.go

```diff
@@ -0,0 +1,206 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package advisor
+
+import (
+	"strings"
+	"time"
+
+	"github.com/boltdb/bolt"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var cmdBucketKey = []byte("Commands")
+
+type writer struct {
+	db *bolt.DB
+	tx *bolt.Tx
+	b  *bolt.Bucket
```

Comment:
Maybe `b` for bucket is a bit too terse? When reading the code below I had to get back here to see what the `b` stands for.